### PR TITLE
chore: Remove `prepublishOnly` script

### PR DESCRIPTION
**Pull request by olivermrbl**
The script is redundant because we always run a separate build step in our release pipelines.

### CODE REVIEW

File: packages/core/medusa-test-utils/package.json [hunk 1]
```diff
@@ -12,7 +12,6 @@
     "node": ">=16"
   },
   "scripts": {
-    "prepublishOnly": "cross-env NODE_ENV=production tsc --build",
     "prepare": "cross-env NODE_ENV=production yarn run build",
```
Comment (adrien2p) [inline packages/core/medusa-test-utils/package.json]: Should we rm this as well?

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: packages/modules/providers/auth-emailpass/package.json [hunk 2]
```diff
@@ -17,10 +17,9 @@
   "author": "Medusa",
   "license": "MIT",
   "scripts": {
-    "prepublishOnly": "cross-env NODE_ENV=production tsc --build",
     "test": "jest --passWithNoTests src",
     "test:integration": "jest --forceExit -- integration-tests/**/__tests__/**/*.spec.ts",
-    "build": "rimraf dist && tsc -p ./tsconfig.json",
+    "build": "rimraf dist && tsc --build ./tsconfig.json",
```
Comment (adrien2p) [inline packages/modules/providers/auth-emailpass/package.json]: Suggestion: The path is not needed here as it is the default

File: packages/modules/providers/auth-google/package.json [hunk 3]
```diff
@@ -17,10 +17,9 @@
   "author": "Medusa",
   "license": "MIT",
   "scripts": {
-    "prepublishOnly": "cross-env NODE_ENV=production tsc --build",
     "test": "jest --passWithNoTests src",
     "test:integration": "jest --forceExit -- integration-tests/**/__tests__/**/*.spec.ts",
-    "build": "rimraf dist && tsc -p ./tsconfig.json",
+    "build": "rimraf dist && tsc --build ./tsconfig.json",
```
Comment (adrien2p) [inline packages/modules/providers/auth-google/package.json]: Suggestion : same where it is specified and not different that the default config file name

**Review by sradevski — APPROVED** [review]

**Pull request merged**